apps/vaporgui/VaporTable.h

```diff
@@ -165,17 +167,24 @@ public slots:
 // int myVal = vaporTable->GetValue(0,0);
 // double myVal = vaporTable->GetValue(0,0);
 // std::string myVal = vaporTable->GetValue(0,0);
-struct Value{
-	std::string _value;
-
-	template<typename T>
-	operator T() const   //implicitly convert into T
-	{
-	   std::stringstream ss(_value);
-	   T convertedValue;
-	   if ( ss >> convertedValue ) return convertedValue;
-	   else throw std::runtime_error("conversion failed");
-	}
+struct Value
+{
+    std::string _value;
+
+    template<typename T>
+    operator T() const   //implicitly convert into T
+    {
+        std::stringstream ss(_value);
+        T convertedValue;
+        if ( ss >> convertedValue )
+            return convertedValue;
+        else
+        {
+            std::string msg = "conversion failed: " + _value;
```

Comment:
This won't fly either. If <T> is a string the else clause will return a NULL. Moreover there is no error checking performed for by VaporTable::GetValue() - the only user of Value - so the error message recorded by SetErrMsg will never be reported.

Not sure what should be done here.